examples/betaflight-sitl/sensors.py

```diff
@@ -0,0 +1,472 @@
+    transmission to Betaflight SITL via the UDP bridge.
+    
+    Args:
+        world_pos: Position [qw, qx, qy, qz, x, y, z]
+        world_vel: Velocity [vx, vy, vz, wx, wy, wz]
+        accel: Accelerometer [ax, ay, az] in body frame
+        gyro: Gyroscope [wx, wy, wz] in body frame
+        baro: Barometer [altitude]
+        
+    Returns:
+        Dictionary with formatted sensor data for FDM packet
+    """
+    # Extract quaternion (Elodin format: [qw, qx, qy, qz])
+    quat = world_pos[:4]
+    position = world_pos[4:7]
+    velocity = world_vel[:3]
+    
+    # Barometric pressure from altitude
+    # Standard atmosphere: P = P0 * (1 - L*h/T0)^(g*M/(R*L))
+    # Simplified: P ≈ 101325 - 12 * h (good for low altitudes)
+    altitude = baro[0] if len(baro) > 0 else position[2]
+    pressure = 101325.0 - 12.0 * altitude
+    
+    return {
+        "angular_velocity": gyro,        # Body frame, rad/s
+        "linear_acceleration": accel,     # Body frame, m/s^2
+        "orientation_quat": quat,         # [w, x, y, z]
+        "velocity": velocity,             # World frame, m/s (ENU)
+        "position": position,             # World frame, m (ENU)
+        "pressure": pressure,             # Pascals
+    }
+
+
+def build_fdm_from_components(
+    world_pos: np.ndarray,
+    world_vel: np.ndarray,
+    accel: np.ndarray,
+    gyro: np.ndarray,
+    timestamp: float,
+    gravity: float = 9.80665,
+) -> "FDMPacket":
+    """
+    Build an FDM packet directly from Elodin component data.
+    
+    This is the primary function for extracting sensor data from the simulation
+    and packaging it for Betaflight. It handles:
+    - Quaternion extraction from world_pos
+    - Velocity extraction from world_vel  
+    - ENU to NED coordinate conversion for gyro/accel
+    - Pressure calculation from altitude
+    
+    Args:
+        world_pos: Position array [qw, qx, qy, qz, x, y, z]
+        world_vel: Velocity array [wx, wy, wz, vx, vy, vz] (angular first in Elodin)
+        accel: Accelerometer [ax, ay, az] in body frame (ENU)
+        gyro: Gyroscope [wx, wy, wz] in body frame (ENU)
+        timestamp: Simulation time in seconds
+        gravity: Gravity constant (for reference)
+        
+    Returns:
+        FDMPacket ready for transmission to Betaflight
+    """
+    # Import here to avoid circular dependency
+    from comms import FDMPacket
+    
+    # Extract quaternion [w, x, y, z]
+    quat = np.array(world_pos[:4])
+    
+    # Extract position [x, y, z] (ENU)
+    position = np.array(world_pos[4:7])
+    
+    # Elodin stores world_vel as [wx, wy, wz, vx, vy, vz]
+    angular_vel = np.array(world_vel[:3])
+    linear_vel = np.array(world_vel[3:6])
+    
+    # Use provided sensor readings or compute from velocity
+    accel_enu = np.array(accel) if accel is not None else np.array([0.0, 0.0, gravity])
+    gyro_enu = np.array(gyro) if gyro is not None else angular_vel
```

Comment:
Ruff check is failing here. Looks like we're on the cusp of adding some error to the sensor values but just a line or two of code short.